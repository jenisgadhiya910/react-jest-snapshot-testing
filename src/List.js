import React, { Component } from 'react';

export default class List extends Component {
    render() {

        const listItems = this.props.items ? this.props.items.map((item, index) => (
            <li key={index}>{item}</li>
        )) : [];

        return (
            <ul>
                {listItems}
            </ul>
        )
    }
}