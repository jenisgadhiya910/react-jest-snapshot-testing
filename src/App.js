import React, { Component } from 'react';
import './App.css';
import List from './List';

export default class App extends Component {
  render() {

    const header = this.props.title
      ? <header className="App-header">
          <h1 className="App-title">{this.props.title}</h1>
        </header>
      : null;


    const items1 = ['foo', 'bar', 'baz'];

    const items2 = ['Lorem', 'ipsum', 'dolor'];

    return (
      <div className="App">
        {header}
        <p className="App-intro">
          App Introduction
        </p>
        <h2>First List</h2>
        <List items={items1} />
        <h2>Second List</h2>
        <List items={items2} />
      </div>
    );
  }
}