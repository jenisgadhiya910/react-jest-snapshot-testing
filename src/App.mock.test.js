import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import List from './List';

jest.mock('./List', () => () => (<list/>));

describe('App - Mocking', () => {

  it('matches snapshot with title', () => {
    const tree = renderer
      .create(<App title="Example Title"/>);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when no title', () => {
    const tree = renderer
      .create(<App/>);
    expect(tree).toMatchSnapshot();
  });

});