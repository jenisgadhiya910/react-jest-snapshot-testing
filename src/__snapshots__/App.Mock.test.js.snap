// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - Mocking matches snapshot when no title 1`] = `
<div
  className="App"
>
  <p
    className="App-intro"
  >
    App Introduction
  </p>
  <h2>
    First List
  </h2>
  <list />
  <h2>
    Second List
  </h2>
  <list />
</div>
`;

exports[`App - Mocking matches snapshot with title 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Example Title
    </h1>
  </header>
  <p
    className="App-intro"
  >
    App Introduction
  </p>
  <h2>
    First List
  </h2>
  <list />
  <h2>
    Second List
  </h2>
  <list />
</div>
`;
